src/compas/geometry/transformations/quaternions.py

```diff
@@ -0,0 +1,324 @@
+        return [x for x in q]
+
+
+def quaternion_conjugate(q):
+    """Conjugate of a quaternion.
+
+    Parameters
+    ----------
+    q : list
+        Quaternion as a tuple of four real values [qw,qx,qy,qz].
+
+    Returns
+    -------
+    list
+        A conjugate quaternion.
+
+    Note
+    ----
+    .. _mathworld quaternion conjugate
+    http://mathworld.wolfram.com/QuaternionConjugate.html
+
+    """
+    return [q[0], -q[1], -q[2], -q[3]]
+
+
+# ----------------------------------------------------------------------
+# quaternion_from_matrix(m) - already in compas/geometry/transformations/matrices.py
+# matrix_from_quaternion(q) - already in compas/geometry/transformations/matrices.py
+# ----------------------------------------------------------------------
+
+
+def quaternion_from_euler_angles(e, static=True, axes='xyz'):
+    """Returns a quaternion from Euler angles.
+
+    Parameters
+    ----------
+    euler_angles : list of float
+        Three numbers that represent the angles of rotations about the specified axes.
+    static : bool, optional
+        If True, the rotations are applied to a static frame.
+        If False, the rotations are applied to a rotational frame.
+        Defaults to True.
+    axes : str, optional
+        A three-character string specifying the order of the axes.
+        Defaults to 'xyz'.
+
+    Returns
+    -------
+    list of float
+        Quaternion [qw,qx,qy,qz] as a list of four real values.
+    """
+
+    m = matrix_from_euler_angles(e, static=True, axes='xyz')
+    q = quaternion_from_matrix(m)
+    return q
+
+
+def euler_angles_from_quaternion(q, static=True, axes='xyz'):
+    """Returns Euler angles from a quaternion.
+
+    Parameters
+    ----------
+    quaternion : list of float
+        Quaternion [qw,qx,qy,qz] as a list of four real values.
+    static : bool, optional
+        If True, the rotations are applied to a static frame.
+        If False, the rotations are applied to a rotational frame.
+        Defaults to True.
+    axes : str, optional
+        A three-character string specifying the order of the axes.
+        Defaults to 'xyz'.
+
+    Returns
+    -------
+    list of float
+        Three Euler angles.
+    """
+    m = matrix_from_quaternion(q)
+    e = euler_angles_from_matrix(m, static, axes)
+    return e
+
+# ----------------------------------------------------------------------
+
+
+def quaternion_from_axis_angle(axis, angle):
+    """Returns a quaternion describing a rotation around the given axis by the given angle.
+
+    Parameters
+    ----------
+    axis : list
+        Coordinates [x,y,z] of the rotation axis vector.
+    angle : float
+        Angle of rotation in radians.
+
+    Returns
+    -------
+    quaternion : list
+        Quaternion [qw,qx,qy,qz] as a list of four real values.
+
+    Example
+    -------
+    >>> axis =  [1.0, 0.0, 0.0]
+    >>> angle = math.pi/2
+    >>> q = quaternion_from_axis_angle(axis,angle)
+    >>> print(q)
```

Comment:
As this might not be true on all machines, suggest to use `compas.geometry.basic.allclose`
```suggestion
    >>> allclose(q, [0.707107, 0.707107, 0, 0], tol=1e-05)
    True
```